azureml-minimal/README.md

```diff
@@ -0,0 +1,325 @@
+
+### Create the resources
+
+1. 🎨 Customize your deployment by editing the default values in the `locals.tf` file.
+
+2. 🔐 Add your secret information like keys and passwords into the `values.tfvars.json` file which is not committed and only exists locally.
+
+3. Initiliaze Terraform modules and download provider definitions.
+
+    ```bash
+    terraform init
+    ```
+
+4. Apply the recipe.
+
+    ```bash
+    terraform apply
+    ```
+
+### Deleting resources
+
+1. 🗑️ Run the destroy function to clean up all resources.
+
+    ```
+    terraform destroy
+    ```
+
+## Running on Azure
+
+Create stack to run on Azure
+
+```shell
+# zenml setup
+export STACK_PROFILE="azureml-mlflow"
+
+# azure credentials
+export SUBSCRIPTION_ID=$(terraform output subscription-id)
+export CLIENT_ID=$(terraform output service-principal-client-id)
+export CLIENT_SECRET=$(terraform output service-principal-client-secret)
+export TENANT_ID=$(terraform output service-principal-tenant-id)
+
+# azure resource group
+export RESOURCE_GROUP=$(terraform output resource-group-name)
+export REGION=$(terraform output resource-group-location)
+
+# azure storage
+export AZURE_STORAGE_CONNECTION_STRING=$(terraform output storage-account-connection-string)
+export CONTAINER_PATH=$(terraform output blobstorage-container-path)
+
+# azureml
+export WORKSPACE_NAME=$(terraform output azureml-workpsace-name)
+export CLUSTER_NAME=$(terraform output azureml-compute-cluster-name)
+export KEY_VAULT_NAME=$(terraform output key-vault-name)
+
+# azure mysql
+export MYSQL_USERNAME=$(terraform output metadata-db-username)
+export MYSQL_PASSWORD=$(terraform output metadata-db-password)
+export MYSQL_SERVER_NAME=$(terraform output metadata-db-host)
+
+# azure mlflow
+export TRACKING_URI=$(terraform output mlflow-tracking-URL)
+
+TOKEN="$(curl -i -X POST \
+  -d "client_id=${CLIENT_ID}" \
+  -d "client_secret=${CLIENT_SECRET}" \
+  -d "grant_type=client_credentials" \
+  -d "resource=https://management.azure.com" \
+  "https://login.microsoftonline.com/${TENANT_ID}/oauth2/token"
+  | jq -r .access_token)"
+
+echo $TOKEN
+
+zenml clean
+zenml init
+zenml profile create $STACK_PROFILE
+zenml profile set azure-mlflow
+
+zenml artifact-store register azure_store \
+    --flavor=azure \
+    --path=$CONTAINER_PATH
+
+zenml secrets-manager register azure_secrets_manager \
+    --flavor=azure_key_vault \
+    --key_vault_name=$KEY_VAULT_NAME
+
+zenml experiment-tracker register mlflow_experiment_tracker --flavor=mlflow --tracking_uri=$TRACKING_URI --tracking_token=$TOKEN
+
+
+zenml metadata-store register azure_mysql \
+    --flavor=mysql \
+    --database=zenml \
+    --secret=azureauthentication \
+    --host=$MYSQL_SERVER_NAME
+
+zenml step-operator register azureml \

```

Comment:
Thanks for writing these steps here too. However, can you add a line towards that says: the recommended approach to register all the resources we deployed here, as a ZenML stack is to import the generated YAML file. However, since we are fetching the token after registration, we have to now update our experiment tracking stack component to include it.